src/main/java/com/iexec/core/chain/DealWatcherService.java

```diff
@@ -175,7 +190,36 @@ void replayDealEvent() {
     private Disposable subscribeToDealEventInRange(BigInteger from, BigInteger to) {
         log.info("Replay Watcher DealEvent started [from:{}, to:{}]",
                 from, (to == null) ? "latest" : to);
-        return iexecHubService.getDealEventObservable(from, to)
+        EthFilter filter = createDealEventFilter(from, to);
+        return iexecHubService.getDealEventObservable(filter)
+                .map(this::schedulerNoticeToDealEvent)
                 .subscribe(dealEvent -> dealEvent.ifPresent(this::onDealEvent));
     }
+
+    EthFilter createDealEventFilter(BigInteger from, BigInteger to) {
+        DefaultBlockParameter fromBlock = DefaultBlockParameter.valueOf(from);
+        DefaultBlockParameter toBlock = DefaultBlockParameterName.LATEST;
+        if (to != null) {
+            toBlock = DefaultBlockParameter.valueOf(to);
+        }
+        EthFilter filter = new EthFilter(fromBlock, toBlock, chainConfig.getHubAddress());
+        BigInteger poolAddressBigInt = Numeric.toBigInt(chainConfig.getPoolAddress());
+        filter.addSingleTopic(EventEncoder.encode(SCHEDULERNOTICE_EVENT));
+        filter.addSingleTopic(Numeric.toHexStringWithPrefixZeroPadded(poolAddressBigInt, 64));
+        return filter;
+    }
+
+    Optional<DealEvent> schedulerNoticeToDealEvent(IexecHubContract.SchedulerNoticeEventResponse schedulerNotice) {
+        dealEventsCount++;
```

Comment:
Isn't your counter is incremented twice (standard + replay)?

Reply:
That is the goal. I want to count all received deal events, even if we receive an event for an already notified event.
It was hard to show in both PRs, but this allowed me a TDD approach and to check the workerpool only receive its specific events.
In the future, when we rework the DealWatcherService, the `dealEventsCount` value will converge to the `receivedDealsCount`.